pkg/build/controller/controller.go

```diff
@@ -326,7 +323,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 
 	// Update the build object when it progress to a next state or the reason for
 	// the current state changed.
-	if build.Status.Phase != nextStatus && !buildutil.IsBuildComplete(build) {
+	if (!hasBuildPodNameAnnotation(build) || build.Status.Phase != nextStatus) && !buildutil.IsBuildComplete(build) {
```

Comment:
In line 335/333 we set the build.Status.Phase to nextPhase.

I'm assuming we don't change anything if build is already complete because there's no other state for the build to go to.